ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/scan/view/ScanView.kt

```diff
@@ -194,22 +196,27 @@ private fun ScanMainContent(
 
     val (scanState, setScanState) = rememberSaveable {
         mutableStateOf(
-            if (permissionState.hasPermission) {
+            if (permissionState.status.isGranted) {
                 ScanState.Scanning
             } else {
                 ScanState.Permission
             }
         )
     }
 
-    if (!permissionState.hasPermission) {
+    if (!permissionState.status.isGranted) {
         setScanState(ScanState.Permission)
-        LaunchedEffect(key1 = UUID.randomUUID()) {
-            permissionState.launchPermissionRequest()
+        if (permissionState.status.shouldShowRationale) {
+            // keep blank screen with a link to the app settings
+            // user denied the permission previously
+        } else {
+            LaunchedEffect(key1 = true) {
```

Comment:
I have a concern this constant key might introduce a different bug.  Did you run through our manual test cases for this?

Reply:
I've been testing several corner cases, see the first attached screen record below. Current implementation seems to be working well imho. Can you see any unwanted behaviour there? On the second attached screen record  I've tried to cover the cases where we probably want to show the "middle state" rational popup, as we've talked about yesterday. I also improved manual test for this screen, see last two commits. Thank you. 

https://user-images.githubusercontent.com/9017390/191736147-7f92188b-f8c7-4c93-aaf8-c5493c752624.mp4


https://user-images.githubusercontent.com/9017390/191736379-a325abd6-ecac-4e6a-b3cc-c67434e88ff9.mp4

Reply:
These videos really help—thanks for capturing them!